quix-frontend/client/src/lib/sql-autocomplete/adapter/sql-autocpmplete-adapter-utills.ts

```diff
@@ -0,0 +1,304 @@
+import { TableInfo } from "../sql-context-evaluator";
+
+interface ObjectChild {
+  name: string;
+  dataType: any;
+}
+
+export function getObjectChildren(obj: Record<string, any>, parentName = ''): ObjectChild[] {
+  const children: ObjectChild[] = [];
+
+  for (const key in obj) {
+    if (typeof obj[key] === 'object' && obj[key] !== null) {
+      const childName = parentName ? `${parentName}.${key}` : key;
+      children.push({ name: childName, dataType: obj[key] });
+      children.push(...getObjectChildren(obj[key], childName));
+    } else {
+      const childName = parentName ? `${parentName}.${key}` : key;
+      children.push({ name: childName, dataType: obj[key] });
+    }
+  }
+
+  return children.map(child => {
+    child.name = child.name.replace(/\.dataType$/, '');
+    return child;
+  });
+}
+
+const enum StartingOptions {
+  row = "row(",
+  map = "map(",
+  array = "array(",
+  timestamp = "timestamp(",
+}
+
+const enum SpecialCharacters {
+  openParenthesis = "(",
+  closedParenthesis = ")",
+  comma = ',',
+  space = ' ',
+}
+
+export function trinoToJs(trinoObjectAsString: string, index: number):  object|string {
```

Comment:
Let's create a class that wrap all trinoToJs logic (let's say `TrinoParser`)..
This class will implement an interface (let's say `ITrinoParser`) with single function `parse`

In that way it will be easier to extend the functionallty with a wrapping class (as proxy) that will cache parsing (read about [Proxy design pattern](https://refactoring.guru/design-patterns/proxy), here is a good example for [caching using Proxy DP](https://refactoring.guru/design-patterns/proxy#:~:text=a%20service%20object.-,Pseudocode,-This%20example%20illustrates))

cc @EvgenyAlterman 
consider it as next task

Reply:
save for l8er